### DISCUSSION TITLE
Is there any plan to improve the software renderer?

### DISCUSSION THREAD

**Discussion by Horbin-Magician** [slint-ui/slint#6038] [Ideas] [answered]
The software renderer of Slint still have some features haven’t been implemented yet, such as:

1. No support for Path.
2. No image rotation or smooth scaling.
3. Text rendering currently limited to western scripts.

Is there any plan to improve them?

**Comment by aurindam** [comment, marked as answer]
Hi,

Yes, we plan to improve the software renderer. There are some tasks planned in the mid-term (6 months) roadmap. Is this something that you need urgently? Are you using the software renderer on an embedded device?

**Reply by Horbin-Magician** [reply]
Thanks for your response.

I'm trying to use the software renderer in PC as an alternate option, because other renderers all (except vulkan, which I can't use) have some incompatibility problem. As a simple UI program, maybe software renderer is acceptable?

I would like software renderer to have the above features especially for Chinese text rendering, but there is no urgency.
Very happy that software renderer is still improving!

**Reply by igiona** [reply]
We're a evaluating this library for a commercial product, for an embedded device without GPU but running on Linux with linuxfb available.
These missing features are kind of crucial for the decision-making process.
Please let me know if there is anything which you would want to test.
